### ISSUE TITLE
`CBaseCSGrenade.GrenadeCount` is always 1

### ISSUE THREAD

**Issue by in0finite** [saul/demofile-net#71]
### Research

- [X] I've already searched and I could not find an existing issue or discussion about this issue.

### Description

`CBaseCSGrenade.GrenadeCount` is always 1, meaning you can't detect if player has 2 flashbangs.

I tried reading Clip1, Clip2, other entries in ReserveAmmo array, and checking if there are multiple CFlashbang entities.

It happens in all demos that I tried (~ 15 of them).

### Code to reproduce

```csharp
var path = ...;
        var demo = new DemoParser();
        var stream = new MemoryStream(File.ReadAllBytes(path));

        await demo.StartReadingAsync(stream, default);

        while (await demo.MoveNextAsync(default))
        {
            var pawns = demo.Players
                .Where(p => p.Pawn != null && p.Pawn.IsAlive)
                .Select(p => (CCSPlayerPawn)p.Pawn!);
               
            var weapons = pawns
                .SelectMany(p => p.Weapons)
                .Where(w => w.ReserveAmmo.Length > 1)
                .Where(w => w.ReserveAmmo[1] > 1)
                .OfType<CBaseCSGrenade>()
                .ToArray();

            if (weapons.Length == 0)
                continue;

            Console.WriteLine($"[{demo.CurrentDemoTick.Value}]\n{string.Join("\n", weapons.Select(w => $"{w.OwnerEntityHandle.Value}, {w.AttributeManager?.Item?.Name}, clip {w.Clip1} {w.Clip2}, ammo {w.ReserveAmmo.ElementAtOrDefault(0)} {w.ReserveAmmo.ElementAtOrDefault(1)} {w.ReserveAmmo.ElementAtOrDefault(2)}"))}\n\n");
            Console.Out.Flush();
        }

        Console.WriteLine("\nFinished!");
```

**Comment by in0finite** [comment]
Found it. Grenade counts are stored in `CCSPlayerPawn.WeaponServices.Ammo[13..17]`.

Why those indexes, I have no idea, tried to search through item definitions, found nothing. It's probably hardcoded (as `AMMO_TYPE` enum). Also, all other entries in `Ammo` array seem to be 0.

Indexes are as following: 
HE : 13
Flash : 14
Smoke : 15
Molotov/Incendiary : 16

Tested against demos from November last year and from few days ago.

Inventory example should probably be updated to display correct values.

**Comment by in0finite** [comment]
Here's quick example to test it :
```cs
var path = ...;

var demo = new DemoParser();

demo.Source1GameEvents.RoundFreezeEnd += e =>
{
    Console.WriteLine($"RoundFreezeEnd\n");

    DumpInventory(demo);

    Console.Out.Flush();
};

var stream = new MemoryStream(File.ReadAllBytes(path));

await demo.ReadAllAsync(stream, default);


    static void DumpInventory(DemoParser demo)
    {
        var pawns = demo.Players
            .Where(p => p.Pawn != null && p.Pawn.IsAlive)
            .Select(p => (CCSPlayerPawn)p.Pawn!)
            //.Where(p => p.Weapons.OfType<CFlashbang>().Any())
            .ToArray();

        if (pawns.Length == 0)
            return;

        Console.WriteLine($"[{demo.TeamCounterTerrorist.Score} - {demo.TeamTerrorist.Score}]\n{string.Join("\n", pawns.Select(p => $"{p.Controller?.PlayerName.PadRight(20)}  {string.Join(", ", p.WeaponServices!.Ammo[13..17])}"))}\n\n");

    }
```

**Comment by saul** [comment]
Thanks for reporting. This will be fixed in v0.16.2
